store/details_v2_test.go

```diff
@@ -305,3 +306,83 @@ func (s *detailsV2Suite) TestInfoFromStoreSnap(c *C) {
 	x := reflect.ValueOf(info).Elem()
 	checker("", x)
 }
+
+// arg must be a pointer to a struct
+func fillStruct(a interface{}, c *C) {
+	va := reflect.ValueOf(a)
```

Comment:
maybe add a quick check for struct then:
```
if t := reflect.TypeOf(); t.Kind() != reflect.Ptr || t.Elem().Kind() != reflect.Struct {
    panic("expected a pointer to a struct")
}
```

Reply:
Fair enough. I thought the current `call of reflect.Value.Elem on struct Value` was clear enough, but it probably needs more chumminess with reflect than can be assumed